README.md
# @capitual/react-native-capface-sdk

Capface sdk adapter to react native. 📱

- [Installation](#installation)
- [Usage](#usage)
- [API](#api)
  - [`initialize(init: CapfaceSdkProps.Initialize)`](#initializeinit-capfacesdkinitialize)
  - [`faceMatch(type: CapfaceSdkProps.MatchType, data?: CapfaceSdkProps.MatchData)`](#facematchtype-capfacesdkmatchtype-data-capfacesdkmatchdata)
  - [`photoMatch(data?: Object)`](#photomatchdata-capfacesdkdata)
  - [`setTheme(options?: CapfaceSdkProps.Theme)`](#setthemeoptions-capfacesdktheme)
- [Types](#types)
  - [`CapfaceSdkProps.Params`](#capfacesdkparams)
  - [`CapfaceSdkProps.Headers`](#capfacesdkheaders)
  - [`CapfaceSdkProps.Theme`](#capfacesdktheme)
  - [`CapfaceSdkProps.ButtonLocation`](#capfacesdkbuttonlocation)
  - [`CapfaceSdkProps.StatusBarColor`](#capfacesdkstatusbarcolor-ios-only)
  - [`CapfaceSdkProps.FeedbackBackgroundColor`](#capfacesdkfeedbackbackgroundcolor-ios-only)
  - [`CapfaceSdkProps.Point`](#capfacesdkpoint-ios-only)
  - [`CapfaceSdkProps.DefaultMessage`](#capfacesdkdefaultmessage)
  - [`CapfaceSdkProps.DefaultScanMessage`](#capfacesdkdefaultscanmessage)
  - [`CapfaceSdkProps.Errors`](#capfacesdkerrors)
  - [`CapfaceSdkProps.MatchType`](#capfacesdkmatchtype)
  - [`CapfaceSdkProps.MatchData`](#capfacesdkmatchdata)
- [Native Events](#native-events)
  - [`Event Types`](#event-types)
- [How to add images in CapfaceSDK module?](#how-to-add-images-in-capfacesdk-module)
  - [How to add images in Android?](#how-to-add-images-in-android)
  - [How to add images in iOS?](#how-to-add-images-in-ios)
  - [Example with images added](#example-with-images-added)
- [Limitations and Features](#limitations-and-features)
- [Contributing](#contributing)
- [License](#license)

<hr/>

## Installation

```sh
npm install @capitual/react-native-capface-sdk
```

<hr/>

## Usage

```tsx
import * as React from 'react';

import {
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
  ScrollView,
  NativeEventEmitter,
} from 'react-native';
import CapfaceSdk, {
  CapfaceSdkProps,
  initialize,
  faceMatch,
  photoMatch,
} from '@capitual/react-native-capface-sdk';

export default function App() {
  const init = async () => {
    /*
     * The SDK must be initialized first
     * so that the rest of the library
     * functions can work!
     *
     * */
    const headers = {
      'clientInfo': 'YUOR_CLIENT_INFO',
      'contentType': 'YOUR_CONTENT_TYPE',
      'device': 'YOUR_DEVICE',
      'deviceid': 'YOUR_DEVICE_ID',
      'deviceip': 'YOUR_DEVICE_IP',
      'locale': 'YOUR_LOCALE',
      'xForwardedFor': 'YOUR_X_FORWARDED_FOR',
      'user-agent': 'YOUR_USER_AGENT',
    };
    const params = {
      device: 'YOUR_DEVICE',
      url: 'YOUR_BASE_URL',
      key: 'YOUR_KEY',
      productionKey: 'YOUR_PRODUCTION_KEY',
    };

    const isInitialized = await initialize({
      params,
      headers,
    });

    console.log(isInitialized);
  };

  const emitter = new NativeEventEmitter(CapfaceSdk);
  emitter.addListener('onCloseModal', (event: boolean) =>
    console.log('onCloseModal', event)
  );

  const onPressPhotoMatch = async () => {
    try {
      const isSuccess = await photoMatch();
      console.log(isSuccess);
    } catch (error: any) {
      console.error(error.message);
    }
  };

  const onPressFaceMatch = async (
    type: CapfaceSdkProps.MatchType,
    data?: CapfaceSdkProps.MatchData
  ) => {
    try {
      const isSuccess = await faceMatch(type, data);
      console.log(isSuccess);
    } catch (error: any) {
      console.error(error.message);
    }
  };

  return (
    <ScrollView style={styles.container}>
      <View style={styles.content}>
        <TouchableOpacity style={styles.button} onPress={init}>
          <Text style={styles.text}>Init Capface Module</Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.button} onPress={onPressPhotoMatch}>
          <Text style={styles.text}>Open Photo Match</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={async () => await onPressFaceMatch('enroll')}
        >
          <Text style={styles.text}>Open Enroll</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={async () =>
            await onPressFaceMatch('authenticate', { id: '123456' })
          }
        >
          <Text style={styles.text}>Open Authenticate</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={async () => await onPressFaceMatch('liveness')}
        >
          <Text style={styles.text}>Open Liveness</Text>
        </TouchableOpacity>
      </View>
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
  },
  content: {
    justifyContent: 'center',
    alignItems: 'center',
    marginVertical: 30,
  },
  button: {
    width: '100%',
    backgroundColor: '#4a68b3',
    padding: 20,
    borderRadius: 15,
    alignItems: 'center',
    justifyContent: 'center',
    marginVertical: 20,
  },
  text: {
    color: 'white',
    fontWeight: '700',
    fontSize: 22,
  },
});
```

<hr/>

## API

| Methods                                                                                                                                      | Return Type        | iOS | Android |
| -------------------------------------------------------------------------------------------------------------------------------------------- | ------------------ | --- | ------- |
| [`initialize(init: CapfaceSdkProps.Initialize)`](#initializeinit-capfacesdkinitialize)                                                       | `Promise<boolean>` | ✅  | ✅      |
| [`faceMatch(type: CapfaceSdkProps.MatchType, data?: CapfaceSdkProps.MatchData`](#facematchtype-capfacesdkmatchtype-data-capfacesdkmatchdata) | `Promise<boolean>` | ✅  | ✅      |
| [`photoMatch(data?: Object)`](#photomatchdata-capfacesdkdata)                                                                                | `Promise<boolean>` | ✅  | ✅      |
| [`setTheme(options?: CapfaceSdkProps.Theme)`](#setthemeoptions-capfacesdktheme)                                                              | `void`             | ✅  | ✅      |

### `initialize(init: CapfaceSdkProps.Initialize)`

This is the **principal** method to be called, he must be **called first** to initialize the Capface SDK. If he doens't be called the other methods **don't works!**

| `CapfaceSdkProps.Initialize` | type                                            | Required | Default     |
| ---------------------------- | ----------------------------------------------- | -------- | ----------- |
| `params`                     | [`CapfaceSdkProps.Params`](#capfacesdkparams)   | ✅       | -           |
| `headers`                    | [`CapfaceSdkProps.Headers`](#capfacesdkheaders) | ❌       | `undefined` |

### `faceMatch(type: CapfaceSdkProps.MatchType, data?: CapfaceSdkProps.MatchData)`

This method is called to make enrollment, authenticate and liveness available. The type is required and it must provided to select which flow you are interested.

- **Enrollment**: This method makes a 3D reading of the user's face. But, you must use to **subscribe** user in Capface SDK or in your server.
- **Authenticate**: This method makes a 3D reading of the user's face. But, you must use to **authenticate** user in Capface SDK or in your server.
- **Liveness**: This method makes a 3D reading of the user's face.

| `Object` | type                                                | Required | Default     |
| -------- | --------------------------------------------------- | -------- | ----------- |
| `type`   | [`CapfaceSdkProps.MatchType`](#capfacesdkmatchtype) | ✅       | -           |
| `data`   | [`CapfaceSdkProps.MatchData`](#capfacesdkmatchdata) | ❌       | `undefined` |

### `photoMatch(data?: Object)`

This method make to read from face and documents for user, after comparate face and face documents from user to check veracity.

| `Object` | type     | Required | Default     |
| -------- | -------- | -------- | ----------- |
| `data`   | `Object` | ❌       | `undefined` |

### `setTheme(options?: CapfaceSdkProps.Theme)`

This method must be used to **set** the **theme** of the Capface SDK screen.

| `CapfaceSdkProps.Theme` | type                                        | Required | Default     |
| ----------------------- | ------------------------------------------- | -------- | ----------- |
| `options`               | [`CapfaceSdkProps.Theme`](#capfacesdktheme) | ❌       | `undefined` |

<hr/>

## Types

| `CapfaceSdkProps` - Types                                                                | iOS | Android |
| ---------------------------------------------------------------------------------------- | --- | ------- |
| [`CapfaceSdkProps.Params`](#capfacesdkparams)                                            | ✅  | ✅      |
| [`CapfaceSdkProps.Headers`](#capfacesdkheaders)                                          | ✅  | ✅      |
| [`CapfaceSdkProps.Theme`](#capfacesdktheme)                                              | ✅  | ✅      |
| [`CapfaceSdkProps.ButtonLocation`](#capfacesdkbuttonlocation)                            | ✅  | ✅      |
| [`CapfaceSdkProps.StatusBarColor`](#capfacesdkstatusbarcolor-ios-only)                   | ✅  | ❌      |
| [`CapfaceSdkProps.FeedbackBackgroundColor`](#capfacesdkfeedbackbackgroundcolor-ios-only) | ✅  | ❌      |
| [`CapfaceSdkProps.Point`](#capfacesdkpoint-ios-only)                                     | ✅  | ❌      |
| [`CapfaceSdkProps.DefaultMessage`](#capfacesdkdefaultmessage)                            | ✅  | ✅      |
| [`CapfaceSdkProps.Errors`](#capfacesdkerrors)                                            | ✅  | ✅      |
| [`CapfaceSdkProps.MatchType`](#capfacesdkdefaultscanmessage)                             | ✅  | ✅      |
| [`CapfaceSdkProps.MatchData`](#capfacesdkdefaultscanmessage)                             | ✅  | ✅      |

### `CapfaceSdkProps.Params`

Here must be passed to initialize the Capface SDK! Case the parameters isn't provided the Capface SDK goes to be not initialized.

| `CapfaceSdkProps.Params` | type      | Required |
| ------------------------ | --------- | -------- |
| `device`                 | `string`  | ✅       |
| `url`                    | `string`  | ✅       |
| `key`                    | `string`  | ✅       |
| `productionKey`          | `string`  | ✅       |
| `isDeveloperMode`        | `boolean` | ❌       |

### `CapfaceSdkProps.Headers`

Here you can add your headers to send request when some method is called. Only values from type **string**, **null** or **undefined** are accepts!

| `CapfaceSdkProps.Headers` | type                            | Required | Default     |
| ------------------------- | ------------------------------- | -------- | ----------- |
| `[key: string]`           | `string`, `null` or `undefined` | ❌       | `undefined` |

### `CapfaceSdkProps.Theme`

This is a list of theme properties that can be used to styling. We recommend that you use hexadecimal values for colors. RGB, RGBA, HSL and HSLA colors are also supported.

| `CapfaceSdkProps.Theme`                        | type                                                                                                                                    | iOS | Android | Required | Default                                                                                                 |
| ---------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------- | --- | ------- | -------- | ------------------------------------------------------------------------------------------------------- |
| `logoImage`                                    | `string`                                                                                                                                | ✅  | ✅      | ❌       | `facetec_your_app_logo.png`                                                                             |
| `cancelImage`                                  | `string`                                                                                                                                | ✅  | ✅      | ❌       | `facetec_cancel.png`                                                                                    |
| `cancelButtonLocation`                         | [`CapfaceSdkProps.ButtonLocation`](#capfacesdkbuttonlocation)                                                                           | ✅  | ✅      | ❌       | `TOP_RIGHT`                                                                                             |
| `defaultStatusBarColorIos`                     | [`CapfaceSdkProps.StatusBarColor`](#capfacesdkstatusbarcolor-ios-only)                                                                  | ✅  | ❌      | ❌       | `DARK_CONTENT`                                                                                          |
| `frameCornerRadius`                            | `number`                                                                                                                                | ✅  | ✅      | ❌       | `10` (iOS) and `20` (Android)                                                                           |
| `frameBackgroundColor`                         | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#FFFFFF`                                                                                               |
| `frameBorderColor`                             | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#FFFFFF`                                                                                               |
| `overlayBackgroundColor`                       | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#FFFFFF`                                                                                               |
| `guidanceBackgroundColorsAndroid`              | `string`                                                                                                                                | ❌  | ✅      | ❌       | `#FFFFFF`                                                                                               |
| `guidanceBackgroundColorsIos`                  | `string[]`                                                                                                                              | ✅  | ❌      | ❌       | `["#FFFFFF", "#FFFFFF"]`                                                                                |
| `guidanceForegroundColor`                      | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#272937`                                                                                               |
| `guidanceButtonBackgroundNormalColor`          | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#026FF4`                                                                                               |
| `guidanceButtonBackgroundDisabledColor`        | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#B3D4FC`                                                                                               |
| `guidanceButtonBackgroundHighlightColor`       | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#0264DC`                                                                                               |
| `guidanceButtonTextNormalColor`                | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#FFFFFF`                                                                                               |
| `guidanceButtonTextDisabledColor`              | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#FFFFFF`                                                                                               |
| `guidanceButtonTextHighlightColor`             | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#FFFFFF`                                                                                               |
| `guidanceRetryScreenImageBorderColor`          | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#FFFFFF`                                                                                               |
| `guidanceRetryScreenOvalStrokeColor`           | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#FFFFFF`                                                                                               |
| `ovalStrokeColor`                              | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#026FF4`                                                                                               |
| `ovalFirstProgressColor`                       | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#0264DC`                                                                                               |
| `ovalSecondProgressColor`                      | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#0264DC`                                                                                               |
| `feedbackBackgroundColorsAndroid`              | `string`                                                                                                                                | ❌  | ✅      | ❌       | `#026FF4`                                                                                               |
| `feedbackBackgroundColorsIos`                  | [`CapfaceSdkProps.FeedbackBackgroundColor` ](#capfacesdkfeedbackbackgroundcolor-ios-only)                                               | ✅  | ❌      | ❌       | [`FeedbackBackgroundColor` ](#capfacesdkfeedbackbackgroundcolor-ios-only)                               |
| `feedbackTextColor`                            | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#FFFFFF`                                                                                               |
| `resultScreenBackgroundColorsAndroid`          | `string`                                                                                                                                | ❌  | ✅      | ❌       | `#FFFFFF`                                                                                               |
| `resultScreenBackgroundColorsIos`              | `string[]`                                                                                                                              | ✅  | ❌      | ❌       | `["#FFFFFF", "#FFFFFF"]`                                                                                |
| `resultScreenForegroundColor`                  | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#272937`                                                                                               |
| `resultScreenActivityIndicatorColor`           | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#026FF4`                                                                                               |
| `resultScreenResultAnimationBackgroundColor`   | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#026FF4`                                                                                               |
| `resultScreenResultAnimationForegroundColor`   | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#FFFFFF`                                                                                               |
| `resultScreenUploadProgressFillColor`          | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#026FF4`                                                                                               |
| `idScanSelectionScreenBackgroundColorsAndroid` | `string`                                                                                                                                | ❌  | ✅      | ❌       | `#FFFFFF`                                                                                               |
| `idScanSelectionScreenBackgroundColorsIos`     | `string[]`                                                                                                                              | ✅  | ❌      | ❌       | `["#FFFFFF", "#FFFFFF"]`                                                                                |
| `idScanSelectionScreenForegroundColor`         | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#272937`                                                                                               |
| `idScanReviewScreenForegroundColor`            | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#FFFFFF`                                                                                               |
| `idScanReviewScreenTextBackgroundColor`        | `string`                                                                                                                                | ✅  | ✅      | ❌       | `#026FF4`                                                                                               |

CHANGELOG.md
## [1.0.4](https://github.com/capitual/react-native-capface/compare/v1.0.3...v1.0.4) (2023-09-12)


### Bug Fixes

* error messages ios ([#5](https://github.com/capitual/react-native-capface/issues/5)) ([6103d1c](https://github.com/capitual/react-native-capface/commit/6103d1c0e82749c01fb78658366a1f338ce341ed))

## [1.0.3](https://github.com/capitual/react-native-capface/compare/v1.0.2...v1.0.3) (2023-09-12)


### Bug Fixes

* error messages android ([#4](https://github.com/capitual/react-native-capface/issues/4)) ([eee093a](https://github.com/capitual/react-native-capface/commit/eee093a73ab2dc2829debe493094700952df615c))

## [1.0.2](https://github.com/capitual/react-native-capface/compare/v1.0.1...v1.0.2) (2023-09-06)


### Bug Fixes

* modules  exports and imports ([1256976](https://github.com/capitual/react-native-capface/commit/125697657502899bb25a76dd213ef92a08647aad))

## [1.0.1](https://github.com/capitual/react-native-capface/compare/v1.0.0...v1.0.1) (2023-09-06)


### Bug Fixes

* adjust publish config of access ([387b810](https://github.com/capitual/react-native-capface/commit/387b810cfd160e4a781ae1e537e5cb6b83076506))

# 1.0.0 (2023-09-06)


### Features

* add types and adjust  docs ([8c93ca8](https://github.com/capitual/react-native-capface/commit/8c93ca8152e4ce0bea8aee0392a0d660a9d3ae8f))
* android modules ([#2](https://github.com/capitual/react-native-capface/issues/2)) ([ee1216d](https://github.com/capitual/react-native-capface/commit/ee1216de47b58ef33d21b05c5c68b51f5abbf0e2))
* create initial project ([8e12d44](https://github.com/capitual/react-native-capface/commit/8e12d445a34879a9bab72f8fad0c749902b9646e))
* ios modules ([#3](https://github.com/capitual/react-native-capface/issues/3)) ([d56a0e8](https://github.com/capitual/react-native-capface/commit/d56a0e8eaad4be68d546d8c0f50c7e15e2986510))


ios/Config.swift
//
//  Config.swift
//  ReactNativeCapfaceSdk
//
//  Created by Nayara Dias, Bruno Fialho and Daniel Sansão on 04/09/23.
//  Copyright © 2023 Capitual. All rights reserved.
//

import UIKit
import Foundation
import FaceTecSDK

public class Config {
    private static let CapThemeUtils: ThemeUtils! = ThemeUtils();
    public static var DeviceKeyIdentifier: String!
    public static var BaseURL: String!
    public static var PublicFaceScanEncryptionKey: String!
    public static var ProductionKeyText: String!
    public static var Headers: NSDictionary?
    public static var Theme: NSDictionary?

    public static func setDevice(_ device: String) {
        Config.DeviceKeyIdentifier = device;
    }

    public static func setUrl(_ url: String) {
        Config.BaseURL = url;
    }

    public static func setKey(_ key: String) {
        Config.PublicFaceScanEncryptionKey = key;
    }

    public static func setProductionKeyText(_ keyText: String) {
        Config.ProductionKeyText = keyText;
    }

    public static func setHeaders(_ headers: NSDictionary?) {
        Config.Headers = headers;
    }

    public static func setTheme(_ theme: NSDictionary?) {
        Config.Theme = theme;
    }

    public static func hasConfig() -> Bool {
        return Config.BaseURL != nil && Config.DeviceKeyIdentifier != nil && Config.ProductionKeyText != nil && Config.PublicFaceScanEncryptionKey != nil
    }

    public static func makeRequest(url: String, httpMethod: String) -> URLRequest {
        let endpoint = BaseURL + url;
        let request = NSMutableURLRequest(url: NSURL(string: endpoint)! as URL)
        request.httpMethod = httpMethod;
        request.addValue(DeviceKeyIdentifier, forHTTPHeaderField: "X-Device-Key");
        request.addValue(FaceTec.sdk.createFaceTecAPIUserAgentString(""), forHTTPHeaderField: "X-User-Agent");

        if (httpMethod != "GET") {
            request.addValue("application/json", forHTTPHeaderField: "Content-Type")
        }

        if (Headers != nil) {
            for key in Headers!.allKeys {
                request.addValue(Headers![key] != nil ? Headers![key] as! String : "", forHTTPHeaderField: key as! String);
            }
        }

        return request as URLRequest;
    }

    static func initialize(_ isDeveloperMode: Bool, completion: @escaping (Bool)->()) {
        if (isDeveloperMode) {
            FaceTec.sdk.initializeInDevelopmentMode(deviceKeyIdentifier: DeviceKeyIdentifier, faceScanEncryptionKey: PublicFaceScanEncryptionKey, completion: { initializationSuccessful in
                completion(initializationSuccessful)
            })
        } else {
            FaceTec.sdk.initializeInProductionMode(productionKeyText: ProductionKeyText, deviceKeyIdentifier: DeviceKeyIdentifier, faceScanEncryptionKey: PublicFaceScanEncryptionKey, completion: { initializationSuccessful in
                completion(initializationSuccessful)
            })
        }
    }

    public static func retrieveConfigurationWizardCustomization() -> FaceTecCustomization {
        let securityWatermarkImage: FaceTecSecurityWatermarkImage = .faceTec;

        let defaultCustomization = FaceTecCustomization();

        defaultCustomization.frameCustomization.cornerRadius = CapThemeUtils.handleBorderRadius("frameCornerRadius");
        defaultCustomization.frameCustomization.backgroundColor = CapThemeUtils.handleColor("frameBackgroundColor");
        defaultCustomization.frameCustomization.borderColor = CapThemeUtils.handleColor("frameBorderColor");

        defaultCustomization.overlayCustomization.brandingImage = CapThemeUtils.handleImage("logoImage", defaultImage: "facetec_your_app_logo");
        defaultCustomization.overlayCustomization.backgroundColor = CapThemeUtils.handleColor("overlayBackgroundColor");

        defaultCustomization.guidanceCustomization.backgroundColors = CapThemeUtils.handleSimpleLinearGradient("guidanceBackgroundColorsIos");
        defaultCustomization.guidanceCustomization.foregroundColor = CapThemeUtils.handleColor("guidanceForegroundColor", defaultColor: "#272937");
        defaultCustomization.guidanceCustomization.buttonBackgroundNormalColor = CapThemeUtils.handleColor("guidanceButtonBackgroundNormalColor", defaultColor: "#026ff4");
        defaultCustomization.guidanceCustomization.buttonBackgroundDisabledColor = CapThemeUtils.handleColor("guidanceButtonBackgroundDisabledColor", defaultColor: "#b3d4fc");
        defaultCustomization.guidanceCustomization.buttonBackgroundHighlightColor = CapThemeUtils.handleColor("guidanceButtonBackgroundHighlightColor", defaultColor: "#0264dc");
        defaultCustomization.guidanceCustomization.buttonTextNormalColor = CapThemeUtils.handleColor("guidanceButtonTextNormalColor");
        defaultCustomization.guidanceCustomization.buttonTextDisabledColor = CapThemeUtils.handleColor("guidanceButtonTextDisabledColor");
        defaultCustomization.guidanceCustomization.buttonTextHighlightColor = CapThemeUtils.handleColor("guidanceButtonTextHighlightColor");
        defaultCustomization.guidanceCustomization.retryScreenImageBorderColor = CapThemeUtils.handleColor("guidanceRetryScreenImageBorderColor");
        defaultCustomization.guidanceCustomization.retryScreenOvalStrokeColor = CapThemeUtils.handleColor("guidanceRetryScreenOvalStrokeColor");

        defaultCustomization.ovalCustomization.strokeColor = CapThemeUtils.handleColor("ovalStrokeColor", defaultColor: "#026ff4");
        defaultCustomization.ovalCustomization.progressColor1 = CapThemeUtils.handleColor("ovalFirstProgressColor", defaultColor: "#0264dc");
        defaultCustomization.ovalCustomization.progressColor2 = CapThemeUtils.handleColor("ovalSecondProgressColor", defaultColor: "#0264dc");

        defaultCustomization.feedbackCustomization.backgroundColor = CapThemeUtils.handleCAGradient("feedbackBackgroundColorsIos");
        defaultCustomization.feedbackCustomization.textColor = CapThemeUtils.handleColor("feedbackTextColor");

        defaultCustomization.cancelButtonCustomization.customImage = CapThemeUtils.handleImage("cancelImage", defaultImage: "facetec_cancel");
        defaultCustomization.cancelButtonCustomization.location = CapThemeUtils.handleButtonLocation("cancelButtonLocation");

        defaultCustomization.resultScreenCustomization.backgroundColors = CapThemeUtils.handleSimpleLinearGradient("resultScreenBackgroundColorsIos");
        defaultCustomization.resultScreenCustomization.foregroundColor = CapThemeUtils.handleColor("resultScreenForegroundColor", defaultColor: "#272937");
        defaultCustomization.resultScreenCustomization.activityIndicatorColor = CapThemeUtils.handleColor("resultScreenActivityIndicatorColor", defaultColor: "#026ff4");
        defaultCustomization.resultScreenCustomization.resultAnimationBackgroundColor = CapThemeUtils.handleColor("resultScreenResultAnimationBackgroundColor", defaultColor: "#026ff4");
        defaultCustomization.resultScreenCustomization.resultAnimationForegroundColor = CapThemeUtils.handleColor("resultScreenResultAnimationForegroundColor");
        defaultCustomization.resultScreenCustomization.uploadProgressFillColor = CapThemeUtils.handleColor("resultScreenUploadProgressFillColor", defaultColor: "#026ff4");

        defaultCustomization.securityWatermarkImage = securityWatermarkImage;

        defaultCustomization.idScanCustomization.selectionScreenBackgroundColors = CapThemeUtils.handleSimpleLinearGradient("idScanSelectionScreenBackgroundColorsIos");
        defaultCustomization.idScanCustomization.selectionScreenForegroundColor = CapThemeUtils.handleColor("idScanSelectionScreenForegroundColor", defaultColor: "#272937");
        defaultCustomization.idScanCustomization.reviewScreenForegroundColor = CapThemeUtils.handleColor("idScanReviewScreenForegroundColor");
        defaultCustomization.idScanCustomization.reviewScreenTextBackgroundColor = CapThemeUtils.handleColor("idScanReviewScreenTextBackgroundColor", defaultColor: "#026ff4");
        defaultCustomization.idScanCustomization.captureScreenForegroundColor = CapThemeUtils.handleColor("idScanCaptureScreenForegroundColor");
        defaultCustomization.idScanCustomization.captureScreenTextBackgroundColor = CapThemeUtils.handleColor("idScanCaptureScreenTextBackgroundColor", defaultColor: "#026ff4");
        defaultCustomization.idScanCustomization.buttonBackgroundNormalColor = CapThemeUtils.handleColor("idScanButtonBackgroundNormalColor", defaultColor: "#026ff4");
        defaultCustomization.idScanCustomization.buttonBackgroundDisabledColor = CapThemeUtils.handleColor("idScanButtonBackgroundDisabledColor", defaultColor: "#b3d4fc");
        defaultCustomization.idScanCustomization.buttonBackgroundHighlightColor = CapThemeUtils.handleColor("idScanButtonBackgroundHighlightColor", defaultColor: "#0264dc");
        defaultCustomization.idScanCustomization.buttonTextNormalColor = CapThemeUtils.handleColor("idScanButtonTextNormalColor");
        defaultCustomization.idScanCustomization.buttonTextDisabledColor = CapThemeUtils.handleColor("idScanButtonTextDisabledColor");
        defaultCustomization.idScanCustomization.buttonTextHighlightColor = CapThemeUtils.handleColor("idScanButtonTextHighlightColor");
        defaultCustomization.idScanCustomization.captureScreenBackgroundColor = CapThemeUtils.handleColor("idScanCaptureScreenBackgroundColor");
        defaultCustomization.idScanCustomization.captureFrameStrokeColor = CapThemeUtils.handleColor("idScanCaptureFrameStrokeColor");

        return defaultCustomization;
    }

    public static func retrieveLowLightConfigurationWizardCustomization() -> FaceTecCustomization {
        return retrieveConfigurationWizardCustomization();
    }


    public static func retrieveDynamicDimmingConfigurationWizardCustomization() -> FaceTecCustomization {
        return retrieveConfigurationWizardCustomization();
    }

    static var currentCustomization: FaceTecCustomization = retrieveConfigurationWizardCustomization();
    static var currentLowLightCustomization: FaceTecCustomization = retrieveLowLightConfigurationWizardCustomization();
    static var currentDynamicDimmingCustomization: FaceTecCustomization = retrieveDynamicDimmingConfigurationWizardCustomization();
}
